hd/src/indexes.rs

```diff
@@ -409,7 +409,10 @@ pub enum AccountStep {
     #[from(u8)]
     #[from(u16)]
     #[from]
-    Normal(UnhardenedIndex),
+    Normal {
+        /// Unhardened derivation 0-based index value
+        index: UnhardenedIndex,
+    },
```

Comment:
I do not entirely get why do we need this. Is it something suggested by the linter?

This is a breaking change which will stop downstream projects from compiling, and will require a major release. Are we really that much needing it?

Reply:
It was originally suggested by the linter, but if that was a breaking put it back. It's basically caused by the error I mentioned and documented here:
https://github.com/LNP-BP/client_side_validation/issues/54

But also, it's weird, once I put it back, the error is gone, which may also invalidate that issue. Good catch, either way, thank you. I wouldn't want to introduce bad code, nor would I want to waste your time.